nixos/tests/beegfs.nix

```diff
@@ -0,0 +1,117 @@
+import ./make-test.nix ({ pkgs, ... } :
+
+let
+  connAuthFile="beegfs/auth-def.key";
+ 
+  client = { config, pkgs, lib, ... } : {
+    networking.firewall.enable = false;
```

Comment:
BeeGFS module should probably configure firewall, rather than requiring users to configure or disable it. (If this is difficult to implement, it may be left for the future.)

Reply:
It is possible but I'd rather leave that for the future, since there are a least two ways to configure the ports. Either manually via extraConfig for each daemon or shifting the standard ports (connPortShift). Is there a good example module for a case like that?

Reply:
We would not extract port options from `extraConfig`, so we would configure firewall only based on the `connPortShift` (and possibly other port-specific options), but that firewall configuration may be left for the user to override in configuration. I have not found examples of this in Nixpkgs.

Reply:
@orivej see https://github.com/NixOS/nixpkgs/issues/19504#issuecomment-253458728 and below

Reply:
This was respected with `openFirewall` boolean option in various services.